api/src/main/java/org/apache/iceberg/actions/DeleteOrphanFiles.java

```diff
@@ -80,6 +80,19 @@
    */
   DeleteOrphanFiles executeDeleteWith(ExecutorService executorService);
 
+  /**
+   * Passes a table which contains the list of actual files in the table. This skips the directory listing - any
+   * files in the actualFilesTable provided which are not found in table metadata will be deleted. Not compatible
+   * with `location` or `older_than` arguments - this assumes that the provided table of actual files has been
```

Comment:
What if the table had a timestamp? For example, if the table has a `modified_time` then we could apply the `older_than` filter fairly easily rather than making the user handle it.

Reply:
Adding `modified_time` sounds like a good idea as ignoring `older_than` is dangerous. Files just written but not yet committed could be cleaned up, corrupting the table.

Reply:
I would even consider making it required. How many people will filter the locations manually before calling the action?